### PR TITLE
Carlos new pie chart hour visualization by team member  project reports

### PR DESCRIPTION
**Pull request by cgomezhub**
(PRIORITY HIGH) Jae: Add “Hours by Team Member” Visualization to the Projects Reports page (WIP Carlos)

Related PRS (if any):
Previous PR2310
This current is related to Backend PR [#1015](https://github.com/OneCommunityGlobal/HGNRest/pull/1015)

Description
The graph currently displays weekly committed hours by project member. However, it does not show the actual hours worked by member on each project.

Main changes explained:
in getTimeEntriesForSpecifiedProjectadd, add firstName lastName and isActive properties in 'TimeEntry' Schema from 'userProfile' Schema with will be received in the fronted.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as Owner, manager or admin user
5. Go to dashboard => report => type the project => PieChart => pick the charts.
6. check and test the pie charts as required

Requested job:

image

video before the change:

https://www.loom.com/share/7be14c4aec464c5fbb9669a11c4d63b5?sid=586d7ba2-8f94-4a4a-99e6-6725b31d4236

video after the change:

https://www.loom.com/share/478133a664ff48cd832d86abdef3edd4?sid=3222bf49-79f9-4fd8-b1d8-abe4bb0687b2

video after the change (toggle type)

https://www.loom.com/share/b4c44576df94464996eb0ef2321c4528?sid=7a1ebf85-e192-451a-b39b-1de04a6cb891

Note:
You must add a project for at least one user ( set as a poject member)
You must log hours for this member(s) in this project.
Go to dashboard => report => type the project => PieChart => pick the charts
4.See videos above for more info

### CODE REVIEW

**Review by peter6866 — DISMISSED** [review]
Changes work as expected.
image

**Review by HowieMiao — CHANGES_REQUESTED** [review]
Pie chart with toggle type works great, shows all info necessary and easy to work with. Only 2 changes that I think would be helpful are first renaming or formatting "PieCharts" to Pie Chart. As well, adjusting colors of text and toggle button to be more legible in Dark Mode.

image

File: src/components/Reports/ProjectReport/__tests__/ProjectReport.test.js [hunk 1]
```diff
@@ -225,39 +225,36 @@ describe('ProjectReport component', () => {
...
-      screen.findByRole('link', { name: mockWBS.wbsName }).then(linkElement => {
+      const linkElement = await screen.findByRole('link', { name: mockWBS.wbsName });
         expect(linkElement).toBeInTheDocument();
         expect(linkElement).toHaveAttribute(
           'href',
           expect.stringContaining(`/wbs/tasks/${mockWBS._id}/${projectId}/${mockWBS.wbsName}`),
         );
-      });
     } else {
-      screen.findByText(mockWBS.wbsName).then(divElement => {
```
Comment (Parth-tech) [inline src/components/Reports/ProjectReport/__tests__/ProjectReport.test.js:256]: Hey Carlos,
I was not able to grasp the change made at line 256.
Why was this test removed?

**Review by Parth-tech — COMMENTED** [review]
Hey @cgomezhub,

As requested by @HowieMiao could you update the UI for the dark mode so that the font is legible. Currently the `username` and the text `All-time total hours by All Members` is not readable due less contrast between the background and the font color. 
The required functionality is met and codebase seems good as well. I found a test being removed it would be great if you could help me why it was replaced completely with new test. Rest of the codebase looks great.

I can approve this once the UI is refined.
Great work done.

`Image Reference:`
Screenshot 2024-07-03 at 3 35 56 PM
Screenshot 2024-07-03 at 3 36 05 PM
Screenshot 2024-07-03 at 3 37 51 PM
Screenshot 2024-07-03 at 3 38 51 PM

File: src/components/Reports/ProjectReport/__tests__/ProjectReport.test.js (hunk 1 above)
Comment (cgomezhub) [reply src/components/Reports/ProjectReport/__tests__/ProjectReport.test.js]: Hi Parth.
it was fixed, please check it _again._
thanks.

**Comment by cgomezhub** [comment]
> Pie chart with toggle type works great, shows all info necessary and easy to work with. Only 2 changes that I think would be helpful are first renaming or formatting "PieCharts" to Pie Chart. As well, adjusting colors of text and toggle button to be more legible in Dark Mode.
> 
> image

**Review by HowieMiao — CHANGES_REQUESTED** [review]
New changes look great, but still seem to be missing high contrast text colors for usernames on Pie Chart hover as well as Weekly committed hours by member toggle at the bottom of the section

image

**Review by DiyaWadhwani — APPROVED** [review]
Hi Carlos,
Everything works as expected. Thank you for your clear instructions. I have attached a video of the pie chart visible with details below, both in the light mode as well as the dark mode.
Thank you

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/54880903/ccdd61ca-36ca-46af-97a0-305f1fbe53cf

**Review by vishavk1992 — APPROVED** [review]
I have tested this PR with backend PR1015,  pie chart visible with details in both mode light and dark mode.  
PR-2409+1015
PR-2409+1015-1

**Review by kdaga81 — COMMENTED** [review]
image

2nd checkbox label is black in color in dark-mode making it hard to read. It can be changes to white for better visibility.

**Pull request merged**